src/changefile/conventionalCommits.ts

```diff
@@ -0,0 +1,44 @@
+import { ChangeType } from '../types/ChangeInfo';
+
+/**
+ * 1. type
+ * 2. scope
+ * 3. breaking
+ * 4. message
+ */
+const COMMIT_RE = /([a-z]+)(?:\(([a-z\-]+)\))?(!)?: (.+)/;
+
+interface ConventionalCommit {
+  type: string;
+  scope?: string;
+  breaking: boolean;
+  message: string;
+}
+
+interface Change {
+  type: ChangeType;
+  message: string;
+}
+
+export function parseConventionalCommit(commitMessage: string): Change | undefined {
+  const match = commitMessage.match(COMMIT_RE);
+  const data: ConventionalCommit | undefined = match
+    ? { type: match[1], scope: match[2], breaking: !!match[3], message: match[4] }
+    : undefined;
+  return data && map(data);
+}
+
+function map(d: ConventionalCommit): Change | undefined {
+  if (d.breaking) {
+    return { type: 'major', message: d.message };
```

Comment:
There's a special case here, for semver 0.x.y packages. In that scheme breaking changes are a minor bump, and fixes/feats are a patch bump.

npm, dependabot, are aware of that scheme, it it's used in some common packages like react-native.